src/schema/objects/common_principles.yaml

```diff
@@ -46,6 +46,9 @@ dataset:
   description: |
     A set of neuroimaging and behavioral data acquired for a purpose of a particular study.
     A dataset consists of data acquired from one or more subjects, possibly from multiple sessions.
+derivative:
+  display_name: derivative dataset
```

Comment:
IMO, derivative and derivative dataset are not identical. If we want to define the generic term "derivative", then it should apply to files and datasets. Similarly with "raw".